Fix: Strip integer part in "decimals"

aaronfranke opened the pull request:
CC @akien-mga Fixes https://github.com/godotengine/godot/issues/22216 See also https://github.com/godotengine/godot/pull/21425 Suggested milestone: 3.1.

DeuxAlpha commented:
Nice! One thing I'd like to mention is that the precision goes up to maxn-1, as I'm sure you know. To increase the decimal points it checks against, both maxn as well as the doubles within the sd-array have to be increased (i.e. 1.000000005 currently returns 0, but by increasing them by one, returns 9).

aaronfranke commented:
@DeuxAlpha Note that when grabbing the value from a single-precision float, it currently cuts off at 7. But yes, I've changed that to go up to 9 like it did before.

aaronfranke committed:
Fix: Strip integer part in "decimals"

Note: Core only.

akien-mga commented on core/math/math_funcs.cpp:
```diff
@@ -67,17 +67,19 @@ int Math::step_decimals(double p_step) {
 	}
 
-	return maxn;
+	return 0;
```
Should it return 0 if it couldn't find any significant digit before running out of things to compare with?

Maybe it could return `-1` instead, and this would be documented as a failure to process the given number.

aaronfranke replied:
0 is what I would expect, but it's fine if we change it to -1, if that's desired.

akien-mga replied:
Nevermind, I get your point. If it can't be matched in the array, it's beyond the precision that we support, and therefore it *is* 0, i.e. `decimals(1.1)` and `decimals(1.100000000013)` should give the same result.

akien-mga commented:
Why not include the same fix in `modules/mono/glue/Managed/Files/Mathf.cs` as done initially in the other PR?

aaronfranke commented:
@akien-mga [The current C# "Decimals"](https://github.com/godotengine/godot/blob/master/modules/mono/glue/Managed/Files/Mathf.cs#L82) behaves as a decimal counter ("1.2345" returns 4) so I think whoever implemented that must've been confused of what the C++ implementation was. The GDscript and C++ behavior is not present in C# yet. I can't fix code which doesn't exist. This is why I added it to my other PR instead of making a new one, but I made one here since you requested it.

If I changed the C# code in this PR, then it would be basically the same as the other PR except for lacking the adding of the "step_decimals" alias. May as well merge the other one.

akien-mga commented:
Thanks, looks good.
